plugins/woocommerce/client/legacy/js/frontend/a8c-address-autocomplete-service.js

```diff
@@ -0,0 +1,485 @@
+										console.error(
+											'Automattic Address Suggestion has been disabled due to internal service error'
+										);
+									}
+									return [];
+								default:
+									return [];
+							}
+						}
+						if ( Array.isArray( data ) ) {
+							data = data.map( ( item ) => ( {
+								id: item.id,
+								label: item.label,
+								matchedSubstrings: item.matched_substrings,
+							} ) );
+							// Cache the successful result
+							cacheResult( inputValue, country, data );
+							return data;
+						}
+					} catch ( e ) {
+						if ( e.name === 'AbortError' ) {
+							// Ignore abort errors from cancelled requests
+							return [];
+						}
+						console.error(
+							'Error fetching address suggestions:',
+							e
+						);
+						return [];
+					}
+				},
+				300,
+				{ leading: false, trailing: true }
+			);
+			window.wc.addressAutocomplete.registerAddressAutocompleteProvider( {
+				id: key,
+				canSearch: () => {
+					try {
+						if ( permanentlyDisabledServices.includes( key ) ) {
+							return false;
+						}
+
+						// Split JWT into parts
+						const [ , payload ] = value.split( '.' );
+						if ( ! payload ) {
+							permanentlyDisabledServices.push( key );
+							return false;
+						}
+
+						// Decode payload
+						const decodedPayload = JSON.parse( atob( payload ) );
+
+						// Check expiration
+						const currentTime = Math.floor( Date.now() / 1000 );
+						if (
+							! decodedPayload.exp ||
+							decodedPayload.exp < currentTime
+						) {
+							permanentlyDisabledServices.push( key );
+							return false;
+						}
+
+						return true;
+					} catch ( e ) {
+						permanentlyDisabledServices.push( key );
+						return false;
+					}
+				},
+				search: async ( inputValue, country, type ) => {
+					// We need to return early here beacuse canSearch is not always called from search.
+					if ( permanentlyDisabledServices.includes( key ) ) {
+						return [];
+					}
+
+					inputValue = inputValue.trim();
+
+					// Check cache first - bypass debounce for cached results
+					const cachedResult = getCachedResult( inputValue, country );
+					if ( cachedResult !== null ) {
+						return cachedResult;
+					}
+
+					return await debouncedSearch( inputValue, country );
+				},
+				async select( addressId ) {
+					const params = new URLSearchParams( {
+						address_id: addressId,
+						session_id: sessionId,
+						lang: document.documentElement.lang,
+						token: value,
+					} );
+
+					const response = await fetch(
+						`${ selectUrl }?${ params.toString() }`
+					);
+
+					let data = await response.json();
+					// Reset session ID after successful select
+					sessionId = crypto.randomUUID();
+					try {
+						dispatchEvent(
+							new CustomEvent(
+								'wc-address-autocomplete-service-request-durations',
+								{
+									detail: {
+										requestDurations,
+									},
+								}
+							)
+						);
```

Comment:
Upon selection we do trigger this event to be consumed by our perf tracking logic.